src/utils/compressImages.ts

```diff
@@ -0,0 +1,131 @@
+import { ImageCompressionOptions } from '../lib/Sendbird';
+import pxToNumber from './pxToNumber';
+import { Logger } from '../lib/SendbirdState';
+
+interface CompressImageProps {
+  imageFile: File;
+  compressionRate: number;
+  resizingWidth?: number;
+  resizingHeight?: number;
+}
+
+const compressImage = ({
+  imageFile,
+  compressionRate,
+  resizingWidth,
+  resizingHeight,
+}: CompressImageProps): Promise<File> => {
+  const image = document.createElement('img');
+  return new Promise((resolve, reject) => {
+    image.src = URL.createObjectURL(imageFile);
+    image.onerror = reject;
+    image.onload = () => {
+      URL.revokeObjectURL(image.src);
+      const canvas = document.createElement('canvas');
+
+      const originWidth = image.width;
+      const originHeight = image.height;
+      const widthRatio = originWidth / (resizingWidth || originWidth);
+      const heightRatio = originHeight / (resizingHeight || originHeight);
+
+      let targetResizingWidth = resizingWidth || originWidth;
+      let targetResizingHeight = resizingHeight || originHeight;
+
+      // Use the more impactful value, so the original images' ratio won't be broken
+      if (widthRatio > heightRatio) {
+        targetResizingHeight = originHeight / (resizingWidth ? widthRatio : 1);
+      } else if (heightRatio > widthRatio) {
+        targetResizingWidth = originWidth / (resizingHeight ? heightRatio : 1);
+      }
+
+      canvas.width = targetResizingWidth;
+      canvas.height = targetResizingHeight;
+
+      const ctx = canvas.getContext('2d');
+      if (!ctx) {
+        reject(new Error('Failed to get canvas 2d context'));
+        return;
+      }
+
+      ctx.drawImage(image, 0, 0, targetResizingWidth, targetResizingHeight);
+
+      ctx.canvas.toBlob(
+        (blob) => {
+          if (blob) {
+            const file = new File([blob], imageFile.name, { type: imageFile.type });
+            resolve(file);
+          } else {
+            reject(new Error('Failed to compress image'));
+          }
+        },
+        imageFile.type,
+        compressionRate,
+      );
+    };
+  });
+};
+
+export interface CompressImagesProps {
+  files: File[];
+  imageCompression: ImageCompressionOptions;
+  logger?: Logger;
+}
+export const compressImages = async ({
+  files,
+  logger,
+  imageCompression,
+}: CompressImagesProps) => {
+  const { compressionRate } = imageCompression;
+  const resizingWidth = pxToNumber(imageCompression.resizingWidth);
+  const resizingHeight = pxToNumber(imageCompression.resizingHeight);
+
+  if (!(Array.isArray(files) && files.length > 0)) {
+    logger.warning('useImageCompression: There is no file.', files);
+    return;
+  }
+  if (compressionRate < 0 || 1 < compressionRate) {
+    logger.warning('useImageCompression: The compressionRate is not acceptable.', compressionRate);
+    return;
+  }
+
+  const compressedImages: File[] = [];
+  const notCompressedFiles: File[] = [];
+
+  await Promise.all(
```

Comment:
hmm compression could take a long time no? if we selected ten 25mb pdf files, it takes long time before actual send. However, I think there is no better option unless we keep our current sending logic that sends one file at a time not altogether at the same time.

Reply:
hm... yes you are right;;
We are adding pending messages in the `onPending` handler
so it's impossible to add pending messages before compressing images for now..

Reply:
To determine if there is a significant impact in practice, it would be a good idea to conduct tests to see how long it takes to compress a 25MB image.

Promise.all processing is influenced by the time taken for the longest operation, so if the individual compression time is not too long, there shouldn't be a significant issue.